### PR TITLE
root: bump api-extractor & friends + recreate API reports along with previously unsupported packages

### PR DESCRIPTION
**Pull request by Rugvip**
A new version of api-extractor is just out, which now supports the previously unsupported `import()` and `import * as ...` syntax, allowing us to create API reports for all packages that we want.

### CODE REVIEW

**Comment by freben** [comment]
Would be neat to some day add a prettier step to these after generation so we ger isolated from little spacing quirk variations

**Comment by benjdlambert** [comment]
I guess we can probably continue the work in this PR rather than #6380 @OrkoHunter?

**Comment by Rugvip** [comment]
@freben making that this day x), as with many things in the api-extractor script, it's not pretty 😅

**Comment by OrkoHunter** [comment]
> I guess we can probably continue the work in this PR rather than #6380 @OrkoHunter?

Sounds good!

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Let's goooo!!

**Comment by benjdlambert** [comment]
@backstage/techdocs-core / @OrkoHunter need a +1 here.

File: yarn.lock [hunk 1]
```diff
@@ -25172,11 +25145,16 @@ typescript-json-schema@^0.50.1:
     typescript "~4.2.3"
     yargs "^16.2.0"
 
-typescript@^4.0.3, typescript@~4.1.3, typescript@~4.2.3:
+typescript@^4.0.3, typescript@~4.2.3:
   version "4.2.4"
   resolved "https://registry.npmjs.org/typescript/-/typescript-4.2.4.tgz#8610b59747de028fda898a8aef0e103f156d0961"
   integrity sha512-V+evlYHZnQkaz8TRBuxTA92yZBPotr5H+WhQ7bD3hZUndx5tGOa1fuCgeSjxAzM1RiN5IzvadIXTVefuuwZCRg==
 
+typescript@~4.3.2:
```
Comment (benjdlambert) [inline yarn.lock:25153]: @Rugvip does this matter that we're now bringing in two versions of typescript?

Comment (Rugvip) [reply yarn.lock:25153]: This one should end up being installed under the api extractor, so no. In the end I'm trusting the tests tho 😁

**Pull request merged**